unittests/AST/StructuralEquivalenceTest.cpp

```diff
@@ -209,5 +209,206 @@ TEST_F(StructuralEquivalenceTest, WrongOrderOfFieldsInClass) {
   EXPECT_FALSE(testStructuralMatch(get<0>(t), get<1>(t)));
 }
 
+TEST_F(StructuralEquivalenceTest, ParamConst1) {
+  auto t = makeNamedDecls("void foo(int&);",
+                          "void foo(const int&);", Lang_CXX);
+  EXPECT_FALSE(testStructuralMatch(get<0>(t), get<1>(t)));
+}
+
+TEST_F(StructuralEquivalenceTest, ParamConst2) {
+  auto t = makeNamedDecls("void foo(int);",
+                          "void foo(const int);", Lang_CXX);
+  EXPECT_TRUE(testStructuralMatch(get<0>(t), get<1>(t)));
+  // this is equal for some reason
+}
+
+TEST_F(StructuralEquivalenceTest, Throw) {
+  auto t = makeNamedDecls("void foo();",
+                          "void foo() throw();", Lang_CXX);
+  EXPECT_FALSE(testStructuralMatch(get<0>(t), get<1>(t)));
+}
+
+TEST_F(StructuralEquivalenceTest, Noexcept1) {
+  auto t = makeNamedDecls("void foo();",
+                          "void foo() noexcept;", Lang_CXX11);
+  EXPECT_FALSE(testStructuralMatch(get<0>(t), get<1>(t)));
+}
+
+TEST_F(StructuralEquivalenceTest, Noexcept2) {
+  auto t = makeNamedDecls("void foo() throw();",
+                          "void foo() noexcept;", Lang_CXX11);
+  EXPECT_TRUE(testStructuralMatch(get<0>(t), get<1>(t)));
+}
+
+TEST_F(StructuralEquivalenceTest, Noexcept3) {
+  auto t = makeNamedDecls("void foo() throw();",
+                          "void foo() noexcept(false);", Lang_CXX11);
+  EXPECT_FALSE(testStructuralMatch(get<0>(t), get<1>(t)));
+}
+
+TEST_F(StructuralEquivalenceTest, Noexcept4) {
+  auto t = makeNamedDecls("void foo() throw();",
+                          "void foo() noexcept(true);", Lang_CXX11);
+  EXPECT_FALSE(testStructuralMatch(get<0>(t), get<1>(t)));
```

Comment:
This should be `EXPECT_TRUE` since `noexcept(true)` equals to `noexcept`.
Rather than `EXPECT_FALSE` we should make the test `DISABLED_`.

Reply:
`// yes: the expression is not evaluated at check`
I wanted to notice that the current implementation does not consider the given expression at all. If the expression is omitted (`noexcept`) it is equivalent to `throw()` but not if there is any expression (including `true`).

ASTStructuralEquivalence.cpp line 441 checks for `noexcept` and `throw()`:
```
      if ((Spec1 != EST_DynamicNone || Spec2 != EST_BasicNoexcept) &&
          (Spec1 != EST_BasicNoexcept || Spec2 != EST_DynamicNone) &&
           Spec1 != Spec2)
```
I did not want to make additional check for ComputedNoexcept and test for a `true` expression or any other that evaluates to `true`.